packages/openapi-fetch/test/index.test.ts

```diff
@@ -1275,6 +1277,186 @@ describe("client", () => {
     });
   });
 
+  describe("body serialization", () => {
+    const BODY_ACCEPTING_METHODS = [["PUT"], ["POST"], ["DELETE"], ["OPTIONS"], ["PATCH"]] as const;
+    const ALL_METHODS = [...BODY_ACCEPTING_METHODS, ["GET"], ["HEAD"]] as const;
+
+    const fireRequestAndGetBodyInformation = async (options: {
+      bodySerializer?: BodySerializer<unknown>;
+      method: (typeof ALL_METHODS)[number][number];
+      fetchOptions: FetchOptions<any>;
+    }) => {
+      const client = createClient<any>({ baseUrl, bodySerializer: options.bodySerializer });
+      const { getRequest } = useMockRequestHandler({
+        baseUrl,
+        method: "all",
+        path: "/blogposts-optional",
+        status: 200,
+      });
+      await client[options.method]("/blogposts-optional", options.fetchOptions as any);
+
+      const request = getRequest();
+      const bodyText = await request.text();
+
+      return { bodyUsed: request.bodyUsed, bodyText };
+    };
+
+    it.each(ALL_METHODS)("missing body (with body serializer) - %s", async (method) => {
```

Comment:
I think we should definitely use more often `it.each` to have cleaner tests while testing more cases. But in this case isn't it overkill to test every method each time as the logic tested is only about falsy values?

What about a single test `it.each([{ body: undefined, except: false }, { body: 1, except: true }, ...])`

Reply:
> But in this case isn't it overkill to test every method each time as the logic tested is only about falsy values?

There are differences in how GET and HEAD are used in some of the underlying implementation (not in this lib, hopefully) - some clients and servers disallow bodies for them. This includes the tools used in these tests. Having to think about these differences inspired me to rather have a comprehensive test of all methods. If you guys don't see it valuable, I'm OK with dropping it.


> What about a single test it.each([{ body: undefined, except: false }, { body: 1, except: true }, ...])

I usually prefer the assertions to be as explicit as possible, hence the explicit `expect().toBe()`, but I don't mind pulling the expectations into the test arguments, if that's what you guys are used to. Matter of taste really.